server/routes/wallet.js

```diff
@@ -248,10 +248,24 @@ router.post('/public-api/buy-address', handler(async (req, res) => {
   } = req.body
   const processor = await plugins.payment
 
-  let ipAddress = req.headers[process.env.IP_HEADER_PROP_NAME] || ''
+  let ipAddress = ''
+  if (Boolean(process.env.TRUST_PROXY) && process.env.IP_HEADER_PROP_NAME) {
```

Comment:
I think you could just check for `process.env.TRUST_PROXY` without `Boolean(process.env.TRUST_PROXY)`